pkg/klusterlet/clusterinfo/clusterinfo_controller.go

```diff
@@ -329,18 +331,17 @@ func (r *ClusterInfoReconciler) getKubeVendor(product string) (kubeVendor cluste
 	return
 }
 
-func (r *ClusterInfoReconciler) isOpenshift() bool {
-	serverGroups, err := r.KubeClient.Discovery().ServerGroups()
+func (r *ClusterInfoReconciler) isOpenshift() (bool, error) {
```

Comment:
Is it better to move this func to a common package, so it can be used in common?

Reply:
changed to check if OCP via product claim.